CONTRIBUTING.md

```diff
@@ -1,30 +1,252 @@
+tested. For code that requires testing, please look for notes in the original issues, as we will do our best to
+provide ideal, meaningful cases to test.
+
+If you feel that there's a test case that we have not considered, please comment in the
+original issue for the team to see.
+
+
+### Pull requests
+
+Pull requests serve a double purpose:
+1. Share the code with the team. So almost everybody is aware of how the code base is evolving.
+2. Provide an opportunity for improving code quality.
+
+When you think your work is done, push the branch to your repository:
+
+```sh
+git push origin issue-1234-new-feature
+# Start a pull request in GitHub
+```
+
+And
+[create a pull request](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request)
+against `main` (or a feature branch).
+When creating the pull request, provide a description and
+[link with the issue that is being solved](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
+
+Linking the issue has the advantage of automatically closing the related issue when the pull
+request is merged.
+
+
+### Code review
+
+When you open a PR you will see a template in the pull request body. Please read it carefully and fill in the necessary
+information to help the code review process go smoothly.
+
+Once you have sent a PR, the code contributors get notified, and there may be a code
+review. The code review helps to solve implementation, semantic and maintainability issues.
+
+The repository also contains some automated checks such as tests and
+[linting](#solving-linting-issues). For a pull request to be ready for merging it needs to
+**pass automatic checks and have, at least, one positive review**.
+
+During code reviews, there are two prominent roles: the reviewer and the contributor.
+The reviewer acts as the keeper of best-practices and code quality, asking
+clarifying questions, highlighting implementation errors and recommending changes.
+We expect the contributor to take recommendations seriously and be willing to
+implement suggested changes or take some other action instead.
+
+Notice we don't expect the contributors to address **all** the comments, nor
+the reviewer highlight **all** the issues, we hope both take some compromises to provide
+as much value and quality as it fits in the estimated effort.
+
+We don't expect discussions to happen in the pull requests. If there is a disagreement,
+our recommendation is for the contributor to yield to the reviewer and for the reviewer
+to suggest other alternatives.
+
+
+### Merging
+
+Once all automated checks are passing and there is a positive review, the pull request
+can be merged. If you are an external contributor, expect your PR to be merged by
+a core contributor.
+
+
+## Code style
+
+Code in this repository should conform to PEP8 standards. Style/lint checks are run to validate this.
```

Comment:
Should we add the link to PEP8? 
https://peps.python.org/pep-0008/

Reply:
Solved in [a3851c0](https://github.com/Qiskit-Extensions/quantum-serverless/pull/211/commits/a3851c08f0a920a6c3cd05171dc78e44e0c99302)